### PR TITLE
Backendconnection

### PR DESCRIPTION
**Pull request by Carolien989**

### CODE REVIEW

File: Switchboard/public/app/components/connections/LinkedFieldsFactory.spec.js [hunk 1]
```diff
@@ -207,36 +211,51 @@ describe("LinkedFieldsFactory", function () {
                         }
                     ]
                 });
-        linkedFields.getLinkedFieldsStream().$promise.then(function () {
+        linkedFields.getLinkedFieldsStream();
 
-            Object.keys(linkedFields.getLinkedFields()).should
-                    .withMessage("Linked Fields should have added an field")
-                    .have.lengthOf(numberOfLinkedFields);
+        $httpBackend.flush(1);
```
Comment (killje) [inline Switchboard/public/app/components/connections/LinkedFieldsFactory.spec.js]: Make the 1 a variable named FLUSH_SINGLE_ITEM

File: Switchboard/public/app/components/connections/LinkedFieldsFactory.js [hunk 2]
```diff
@@ -277,46 +239,104 @@ var LinkedFieldsClass = (function (
             return changesSaved;
         };
 
-        var fakeIndex = 0;
-
         /**
          * @ngdoc method
          * @name LinkedFieldsFactory#saveChanges
          * @description
+         * @return {Promise} promise of all collected promises }
```
Comment (killje) [inline Switchboard/public/app/components/connections/LinkedFieldsFactory.js]: unneeded } at the end

File: Switchboard/public/app/components/connections/LinkedFieldsFactory.spec.js (hunk 1 above)
Comment (killje) [inline Switchboard/public/app/components/connections/LinkedFieldsFactory.spec.js]: Also remove the eslint disable

File: Switchboard/public/app/components/connections/OverviewFactory.spec.js [hunk 3]
```diff
@@ -14,6 +14,15 @@ describe("OverviewFactory", function () {
...
+    it("getInformation");
+    
+    
+    it("getConnections");
+    
+    
+    it("close");
+    
+    
+    it("click");
```
Comment (killje) [inline Switchboard/public/app/components/connections/OverviewFactory.spec.js]: This is in a trello card?

Comment (Carolien989) [inline Switchboard/public/app/components/connections/OverviewFactory.spec.js]: Ther is a Trello Card for it,  overviewFactory is the only factory who still needs tests

File: Switchboard/public/app/components/connections/LinkedFieldsFactory.spec.js [hunk 4]
```diff
@@ -39,17 +40,19 @@ describe("LinkedFieldsFactory", function () {
 
     beforeEach(angular.mock.inject([
         "$injector",
+        "$timeout",
```
Comment (killje) [inline Switchboard/public/app/components/connections/LinkedFieldsFactory.spec.js]: timeout is not needed anymore

File: Switchboard/public/app/components/connections/LinkedFieldsFactory.spec.js [hunk 5]
```diff
@@ -39,17 +40,19 @@ describe("LinkedFieldsFactory", function () {
...
         "ResourcesFactory",
         "connections/LinkedFieldsFactory",
         function ($injector,
+                $timeoutObject,
```
Comment (killje) [inline Switchboard/public/app/components/connections/LinkedFieldsFactory.spec.js]: Timeout is not needed aymore

**Pull request merged**